matter/wifi/rs911x/rsi_if.c

```diff
@@ -245,71 +250,73 @@ static int32_t wfx_rsi_init(void)
                 return status;
         }
 #endif /* missing in sapi */
-  if ((status = rsi_wlan_register_callbacks(RSI_JOIN_FAIL_CB, wfx_rsi_join_fail_cb)) != RSI_SUCCESS) {
-    WFX_RSI_LOG("%s: RSI callback register join failed with status: %02x", __func__, status);
+  if ((status = rsi_wlan_register_callbacks(
+           RSI_JOIN_FAIL_CB, wfx_rsi_join_fail_cb)) != RSI_SUCCESS) {
+    WFX_RSI_LOG("%s: RSI callback register join failed with status: %02x",
+                __func__, status);
     return status;
   }
 #ifdef RS911X_SOCKETS
-  (void)rsi_wlan_register_callbacks(RSI_IP_CHANGE_NOTIFY_CB, wfx_rsi_ipchange_cb);
+  (void)rsi_wlan_register_callbacks(RSI_IP_CHANGE_NOTIFY_CB,
+                                    wfx_rsi_ipchange_cb);
 #else
-  if ((status = rsi_wlan_register_callbacks(RSI_WLAN_DATA_RECEIVE_NOTIFY_CB, wfx_rsi_wlan_pkt_cb)) != RSI_SUCCESS) {
-    WFX_RSI_LOG("%s: RSI callback register data-notify failed with status: %02x", __func__, status);
+  if ((status = rsi_wlan_register_callbacks(RSI_WLAN_DATA_RECEIVE_NOTIFY_CB,
+                                            wfx_rsi_wlan_pkt_cb)) !=
+      RSI_SUCCESS) {
+    WFX_RSI_LOG(
+        "%s: RSI callback register data-notify failed with status: %02x",
+        __func__, status);
     return status;
   }
 #endif
   wfx_rsi.dev_state |= WFX_RSI_ST_DEV_READY;
   WFX_RSI_LOG("%s: RSI: OK", __func__);
   return RSI_SUCCESS;
 }
-void wfx_show_err(char *msg)
-{
-  WFX_RSI_LOG("%s: message: %d", __func__, msg);
-}
+void wfx_show_err(char *msg) { WFX_RSI_LOG("%s: message: %d", __func__, msg); }
 /*
  * Saving the details of the AP
  */
-static void wfx_rsi_save_ap_info()
-{
+static void wfx_rsi_save_ap_info() {
   int32_t status;
   rsi_rsp_scan_t rsp;
 
-  status = rsi_wlan_scan_with_bitmap_options((int8_t *)&wfx_rsi.sec.ssid[0], 0, &rsp, sizeof(rsp), 1);
+  status = rsi_wlan_scan_with_bitmap_options((int8_t *)&wfx_rsi.sec.ssid[0], 0,
+                                             &rsp, sizeof(rsp), 1);
   if (status) {
     /*
      * Scan is done - failed
      */
   } else {
     wfx_rsi.sec.security = rsp.scan_info->security_mode;
-    wfx_rsi.ap_chan      = rsp.scan_info->rf_channel;
+    wfx_rsi.ap_chan = rsp.scan_info->rf_channel;
     memcpy(&wfx_rsi.ap_mac.octet[0], &rsp.scan_info->bssid[0], 6);
   }
   if ((wfx_rsi.sec.security == RSI_WPA) || (wfx_rsi.sec.security == RSI_WPA2)) {
+    // saving the security before changing into mixed mode
+    security = wfx_rsi.sec.security;
     wfx_rsi.sec.security = RSI_WPA_WPA2_MIXED;
   }
-  if (wfx_rsi.sec.security == 7){
+  if (wfx_rsi.sec.security == 7) {
+    security = 3; // WPA3 expects security as 3
```

Comment:
Could you set a constant or enum or defines for those values with comments maybe. Why 7 en then why 3.

Reply:
updated in the latest commit